pallets/order-book/src/order_book.rs

```diff
@@ -361,23 +497,43 @@ impl<T: crate::Config + Sized> OrderBook<T> {
 
         ensure!(remaining_amount.is_zero(), Error::<T>::NotEnoughLiquidity);
 
-        let (market_input, market_output) = match side {
+        payment
+            .to_lock
+            .entry(maker_out_asset)
+            .or_default()
+            .entry(taker)
+            .and_modify(|lock_amount| *lock_amount += maker_amount)
+            .or_insert(maker_amount);
```

Comment:
Might be handy to make functions like `lock` in `Payment` (and rename existing `lock` to `execute_lock`) and just pass three parameters instead of writing this long expression
not critical ofc

Reply:
Disagree. `lock` is a verb and it is logical that assets are locked immediately when you call `lock` method

Reply:
Makes sense, then something like `schedule_lock` should fit